libtenzir/src/plugin.cpp

```diff
@@ -636,6 +639,84 @@ auto aspect_plugin::aspect_name() const -> std::string {
   return name();
 }
 
+// -- parser plugin ------------------------------------------------------------
+
+auto plugin_parser::parse_strings(std::shared_ptr<arrow::StringArray> input,
+                                  operator_control_plane& ctrl) const
+  -> std::vector<std::pair<type, std::shared_ptr<arrow::Array>>> {
+  // TODO: Collecting finished table slices here is very bad for performance.
+  // For example, we have to concatenate new table slices. But there are also
+  // many questions with regards to semantics. This should be either completely
+  // rewritten or replaced with a different mechanism after the revamp.
```

Comment:
Agree, it looks bad, but if the semantics are at least workable for now, I'd say it's fine.